readme.md: Fix broken link to Doxygen version 1.7.3 Windows installer…

JulienCochuyt opened the pull request:
… (#9513)

<!--
Please fill in the following template, for an explanation of the sections see:
https://github.com/nvaccess/nvda/wiki/Github-pull-request-template-explanation-and-examples
-->

### Link to issue number:

Fixes #9513

### Summary of the issue:

In readme.md, the link to Doxygen is broken: http://www.stack.nl/~dimitri/doxygen/download.html
Returns 403 Forbidden

### Description of how this pull request fixes the issue:

Direct link to Doxygen 1.7.3 Windows installer at SourceForge: https://sourceforge.net/projects/doxygen/files/rel-1.7.3/doxygen-1.7.3-setup.exe

### Testing performed:

This new link is active.
Did not try to install, though.

### Known issues with pull request:

### Change log entry:

N/A

JulienCochuyt committed:
readme.md: Fix broken link to Doxygen version 1.7.3 Windows installer (#9513)

leonardder commented:
We've changed the url to doxygen in the threshold_py3_staging branch and also updated the minimum version. I think it is best to close this for now.

JulienCochuyt commented:
> We've changed the url to doxygen in the threshold_py3_staging branch and also updated the minimum version. I think it is best to close this for now.

Is threshold_py3_staging fully ready yet and about to be merged to master?

leonardder commented:
I think that will still take some time. However, I don't consider this 
issue very major. Having said that, may be @feerrenrut wants to have it 
merged regardless, as it is really a trivial and quick change.
